twake/frontend/src/app/scenes/Client/ChannelsBar/Parts/Channel/ChannelMenu.tsx

```diff
@@ -185,6 +185,11 @@ export default (props: Props): JSX.Element => {
         },
       },
     );
+
+    // This remove the edit channel menu item when user isn't channel owner
+    if (props.channel.owner !== currentUser.id) {
+      menu.splice(4, 1);
```

Comment:
Is it the way to do in react, seriously? Can't we apply rules on each menu item based on the user roles?

Reply:
Something like this ?

```js
{
      type: 'menu',
      rules: props.channel.owner === currentUser.id,
      text: Languages.t('scenes.app.channelsbar.modify_channel_menu'),
      onClick: () => {
        editChannel();
      },
}
...

const filteredMenu = menu.filter((item: any) => item.rules !== false);
```

Reply:
We could add if / else while pushing menu elements, but your proposition @stephanevieira75 is good, I really like it.
Can you:
- rename "rules" to "disabled" (and so inverse the condition)
- put this line 175 of MenusBodyLayer:
```
          {this.state.menus_manager.menus.filter(item=>!item.disabled).map((item, i) => {
```

Reply:
Well, item can be enabled/disabled/hidden, so is `disabled` the right name?

Reply:
We already had a "hide" option in the menu builder, so we used it 👍